### PR TITLE
newMockStore: add

### PR DESCRIPTION
**Pull request by chukarave**
Copy of [0] and test

[0] https://gerrit.wikimedia.org/r/plugins/gitiles/mediawiki/extensions/Wikibase/+/refs/heads/master/client/data-bridge/tests/unit/store/newMockStore.ts

### CODE REVIEW

File: src/newMockStore.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { ActionContext, Commit, Dispatch } from 'vuex';
+
+export default function newMockStore(
```
Comment (AlaaSarhan) [inline src/newMockStore.ts:3]: we can add a docblock here mentioning what can be passed in? or redundant?

**Review by AlaaSarhan — APPROVED** [review]
Looks good!

File: src/newMockStore.ts (hunk 1 above)
Comment (AlaaSarhan) [reply src/newMockStore.ts:3]: hmm probably docs would be redundant in case of typescript with proper tooling, I suppose

Comment (tarrow) [reply src/newMockStore.ts:3]: I think we can get away without a docblock given the ubquity of the structure of a vuex store and the type hinting we take from vuex for the more complex parts. Would you think we need a hint to the vuex docs?

File: src/newMockStore.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		dispatch?: Dispatch;
+		state?: any;
+		getters?: object;
+		rootState?: any;
+		rootGetters?: object;
+	},
+): ActionContext<any, any> {
+	return {
+		commit: commit || jest.fn(),
+		dispatch: dispatch || jest.fn( () => Promise.resolve() ),
```
Comment (wiese) [inline src/newMockStore.ts:15]: Sneaky fix is sneaky.

File: src/newMockStore.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { ActionContext, Commit, Dispatch } from 'vuex';
+
+export default function newMockStore(
+	{ commit, dispatch, state, getters, rootState, rootGetters }: {
+		commit?: Commit;
+		dispatch?: Dispatch;
+		state?: any;
+		getters?: object;
```
Comment (wiese) [inline src/newMockStore.ts:8]: What happened to the `GetterTree` and why?

File: src/newMockStore.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+		rootState?: any;
+		rootGetters?: object;
```
Comment (wiese) [inline src/newMockStore.ts:10]: What happened to the `GetterTree` and why?

**Review by wiese — COMMENTED** [review]
Not the sole fault of this patch but maybe it would be a good moment to add (e.g. in README) some overview what this lib offers and when to use what.

E.g. "newMockStore can be helpful when unit testing store actions and their effects"

File: src/newMockStore.ts (hunk 3 above)
Comment (tarrow) [reply src/newMockStore.ts:8]: Interestingly ActionContext interface only requires any not specifically a GetterTree

Were you thinking we should have instead met the StoreOptions interface?

Comment (wiese) [reply src/newMockStore.ts:8]: I was (almost) not thinking at all but comparing with [`newMockStore` in termbox](https://gerrit.wikimedia.org/r/plugins/gitiles/wikibase/termbox/+/bd69d5a25d8bdd10d8de5ac397c4b5b8b4c3bc33/tests/unit/store/newMockStore.ts#8). Maybe `ActionContext` is sloppy about this (github karma right there) but there is no doubt in my mind that `getters` and `rootGetters` contain a tree of getters...

Comment (jakobw) [reply src/newMockStore.ts:8]: I didn't verify this but the reasoning here makes sense to me: https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/534603

That would also explain why https://github.com/vuejs/vuex/blob/0e109e2a38dafdc0c2bd6bd3892bc66cfe252b16/types/index.d.ts#L55 is not hinting `GetterTree`

**Review by jakobw — APPROVED** [review]

**Review by mariushoch — APPROVED** [review]

**Pull request merged**